template/template.go

```diff
@@ -30,9 +32,18 @@ type Template struct {
 // path. If the template does not exist, an error is returned. During
 // initialization, the template is read and is parsed for dependencies. Any
 // errors that occur are returned.
-func NewTemplate(path, leftDelim, rightDelim string) (*Template, error) {
+func NewTemplate(path, contents, leftDelim, rightDelim string) (*Template, error) {
```

Comment:
I would actually keep the path as the `ID` method relies on it to disambiguate two templates with the same content